PlayerCore/components/New VRM Core/VRMItemResponseTime.swift

```diff
@@ -0,0 +1,78 @@
+//  Copyright 2019, Oath Inc.
+//  Licensed under the terms of the MIT License. See LICENSE.md file in project root for terms.
+
+import Foundation
+
+public struct VRMItemResponseTime {
+    typealias Container = [VRMCore.Item: TimeRange]
+    
+    static let initial = VRMItemResponseTime(timeRangeContainer: [:])
+    
+    public struct TimeRange: Hashable {
+        public let startAt: Date
+        public var finishAt: Date?
+    }
+    
+    public let timeRangeContainer: [VRMCore.Item: TimeRange]
+}
+
+func reduce(state: VRMItemResponseTime, action: Action) -> VRMItemResponseTime {
+    
+    func updateFinishTime(for item: VRMCore.Item,
+                          with date: Date,
+                          in container: VRMItemResponseTime.Container) -> VRMItemResponseTime.Container {
+        guard var range = container[item] else {
+            fatalError("get finish error action for item which wasn't started")
+        }
+        range.finishAt = date
+        var newTimeContainer = container
+        newTimeContainer[item] = range
+        
+        return newTimeContainer
+    }
+    
+    func updateStartTime(for item: VRMCore.Item,
+                         with date: Date) -> VRMItemResponseTime {
+        let range = VRMItemResponseTime.TimeRange(startAt: date, finishAt: nil)
+        var newTimeContainer = state.timeRangeContainer
+        newTimeContainer[item] = range
+        
+        return VRMItemResponseTime(timeRangeContainer: newTimeContainer )
+    }
+    
+    switch action {
+    case let failedFetch as VRMCore.FetchingError:
+        let newContainer = updateFinishTime(for: failedFetch.originalItem,
+                                            with: failedFetch.finishDate,
+                                            in: state.timeRangeContainer)
+        return VRMItemResponseTime(timeRangeContainer: newContainer )
+    case let failedParse as VRMCore.ParsingError:
+        let newContainer = updateFinishTime(for: failedParse.originalItem,
+                                            with: failedParse.finishDate,
+                                            in: state.timeRangeContainer)
+        return VRMItemResponseTime(timeRangeContainer: newContainer )
+    case let indirectionError as VRMCore.TooManyIndirections:
+        let newContainer = updateFinishTime(for: indirectionError.item,
+                                            with: indirectionError.finishDate,
+                                            in: state.timeRangeContainer)
+        return VRMItemResponseTime(timeRangeContainer: newContainer )
```

Comment:
Make sense to move this inside function - no reason to duplicate it each time.

Reply:
I've made this for `HardTimeout` case. There we have array of items.

Reply:
Than for hard timeout write additional code - but 90% of other cases are will be covered.